src/cargo/core/compiler/build_runner/mod.rs

```diff
@@ -302,6 +303,49 @@ impl<'a, 'gctx> BuildRunner<'a, 'gctx> {
                     .insert(dir.clone().into_path_buf());
             }
         }
+
+        if self.bcx.build_config.intent.is_doc()
+            && self.bcx.gctx.cli_unstable().rustdoc_mergeable_info
+            && let Some(unit) = self
+                .bcx
+                .roots
+                .iter()
+                .filter(|unit| unit.mode.is_doc())
+                .next()
+        {
+            let mut rustdoc = self.compilation.rustdoc_process(unit, None)?;
+            let doc_dir = self.files().out_dir(unit);
+            let mut include_arg = OsString::from("--include-parts-dir=");
+            include_arg.push(self.files().doc_parts_dir(&unit));
+            rustdoc
+                .arg("-o")
+                .arg(&doc_dir)
+                .arg("--emit=toolchain-shared-resources")
```

Comment:
Adding `--emit=toolchain-shared-resources` to `--merge=finalize` will hinder `rustdoc` from generating search index. I don't know if it is a bug or intentional, or just `--emit` and `--merge` haven't yet collaborate with each other.

Reply:
That does seem weird. Probably a cut-and-paste error.